src/broker/broker.c

```diff
@@ -3367,3 +3325,65 @@ get_as_slow_log_filename (char *log_filename, int len, char *broker_name, T_APPL
       log_filename[0] = '\0';
     }
 }
+
+static int
+read_buffer_async (SOCKET sock_fd, char *buf, int size, int timeout)
+{
+  int read_len = -1;
+  struct pollfd po[1] = { {0, 0, 0} };
+  int ret;
+
+  po[0].fd = sock_fd;
+  po[0].events = POLLIN;
+
+  do
+    {
+      ret = poll (po, 1, timeout * 1000);
+    }
+  while (ret < 0 && errno == EINTR);
+
+  if (ret < 1)			/* ERROR OR TIMEOUT */
+    {
+      return -1;
+    }
+
+  if (po[0].revents & POLLIN)	/* RECEIVE NEW REQUEST */
+    {
+      read_len = READ_FROM_SOCKET (sock_fd, buf, size);
+    }
+
+  return read_len;
+}
+
+static int
+write_buffer_async (SOCKET sock_fd, char *buf, int size, int timeout)
+{
+  int write_len = -1;
+  struct pollfd po[1] = { {0, 0, 0} };
+  int ret;
+
+#ifdef ASYNC_MODE
```

Comment:
How about moving "#ifdef ASYNC_MODE" from this function to write_to_client_with_timeout() like  implementing "#ifdef ASYNC_MODE" at read_to_client_with_timeout() ?

Reply:
It looks much better.
Thank you for your comment!